src/libAtomVM/otp_net.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * This file is part of AtomVM.
+ *
+ * Copyright 2023 by Fred Dushin <fred@dushin.net>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+ */
+
+#include <context.h>
+#include <defaultatoms.h>
+#include <globalcontext.h>
+#include <interop.h>
+#include <nifs.h>
+#include <otp_net.h>
+#include <port.h>
+#include <term.h>
+
+#include <errno.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+// #define ENABLE_TRACE
+#include <trace.h>
+
+#define FAMILY_ATOM globalcontext_make_atom(global, ATOM_STR("\x6", "family"))
+#define PROTOCOL_ATOM globalcontext_make_atom(global, ATOM_STR("\x8", "protocol"))
+#define TYPE_ATOM globalcontext_make_atom(global, ATOM_STR("\x4", "type"))
+#define INET_ATOM globalcontext_make_atom(global, ATOM_STR("\x4", "inet"))
+#define ADDR_ATOM globalcontext_make_atom(global, ATOM_STR("\x4", "addr"))
+// NB. The erlang documentation states that an address info structure is a map
+// with an `address` key, but OTP-24 returns maps with an `addr` key.
+#define ADDRESS_ATOM globalcontext_make_atom(global, ATOM_STR("\x7", "address"))
+
+#define UNKNOWN_TABLE_VALUE -1
+
+static const AtomStringIntPair protocol_table[] = {
+    { ATOM_STR("\x3", "tcp"), IPPROTO_TCP },
+    { ATOM_STR("\x3", "udp"), IPPROTO_UDP },
+    SELECT_INT_DEFAULT(UNKNOWN_TABLE_VALUE)
+};
+
+static const AtomStringIntPair type_table[] = {
```

Comment:
A very similar table exists in otp_socket.c. Can we factorize somehow?

Reply:
Maybe after merging PR #908 ?

Reply:
To be addressed in PR #920